Assets/Scripts/Managers/UIManager.cs

```diff
@@ -8,17 +8,28 @@ public class UIManager : MonoBehaviour {
 		// [SerializeField] private bool _hasPlaytestKey = false;
 		public bool HasPlaytestKey = false;
 		public float Health = 1;
+		public float Dash = 1;
+
+		private float _speed = 0.3f;
+
+		[HideInInspector] public Inventory PlayerInventory { get; private set; }
+		public GameObject Player;
```

Comment:
Needs to be set in every scene which is frankly quite annoying. Maybe replace with something like `			Player = FindFirstObjectByType<StateMachine>().gameObject;
` in the Awake.
Is there a reason for this and the Inventory object to be public? 
Also why even store the gameobject at all if only the inventory component is actually used?

Reply:
Made the inventory a private, copied this initially from another file and forgot to adjust it.
Tried retrieving the inventory directly from the player with the code you provided and that seems to work as well so the Gameobject player has been deleted